### PR TITLE
{lib}[fosscuda/2019b] TensorFlow v2.3.0 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

- [x] Requires ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2110~~
- [x] Requires working TF_SYSTEMLIBS: ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2117~~, ~~https://github.com/easybuilders/easybuild-easyconfigs/pull/11109~~

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusi4250.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz, Python 2.7.5
See https://gist.github.com/9dc7978f8af2b044320f8078887ebcc7 for a full test report.

**Comment by boegel** [comment]
@Flamefire Can you take care of the missing easyconfig for `Bazel` ?

**Comment by Flamefire** [comment]
Sure. It's missing the easyblock change but I'm still having trouble on power but got a solution in the pipeline. I'll be able to test that next week. 
You could merge the current easyblock PR which allows us to remove a patch from all semi-current ecs

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.0-fosscuda-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+    'sanity_pip_check': True,
+}
+use_pip = True
+
+# Dependecies created and updated using findPythonDeps.sh:
+# https://gist.github.com/Flamefire/49426e502cd8983757bd01a08a10ae0d
+exts_list = [
+    ('scipy', '1.4.1', {
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.0-fosscuda-2019b-Python-3.7.4.eb:54]: @Flamefire We should also remove this in favor of using `scipy` as a dependency?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusml30 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/0a5c9cb080c62c0c7e625fcae98b7207 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusi5597.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/65dd83de29d4968761f9934ec5920a53 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusml2 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/ceb66bd664b1cf077c8d2efa6542dd04 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusi5064.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/932c7dca1f148c10d7c66237b1f7f8f9 for a full test report.

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
freja - Linux Ubuntu 20.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 3.8.2
See https://gist.github.com/056b551e77ec1607af162471c2e55555 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 10 out of 12 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/5b3a32513be3cfb1d183dea2e8c80785 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/d124c7537f2429473a77ab64aaaf8c6a for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
gligar04.gastly.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/4db005f085c9d2dddcf65cba73129cc1 for a full test report.

**Comment by Flamefire** [comment]
@boegel You are likely using the TF EasyBlock without the JSON fix

**Comment by boegel** [comment]
@Flamefire Nevermind that test report, I started it in the wrong session...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
node3413.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/c4bdfb978d5086212dc703442906dc4f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
node3413.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/af860be930c49fd5766c0b2da80d7c76 for a full test report.

**Comment by boegel** [comment]
```
$ python /tmp/easybuild_build/TensorFlow/2.3.0/foss-2019b-Python-3.7.4/TensorFlow-2.x_mnist-test.py"
TypeError: Parameter to MergeFrom() must be instance of same class: expected tensorflow.TensorShapeProto got tensorflow.TensorShapeProto.
```

How does this only happen on some systems and not on others?!

**Comment by Flamefire** [comment]
This is the protobuf bug I fixed last minute. Reinstall the protobuf-python to fix. I short: the c++ extension for protobuf is much faster and wrong

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz (cascadelake), Python 3.6.8
See https://gist.github.com/ced6c60e2ab6b1bd3bf4e426d4e22e02 for a full test report.

**Comment by terjekv** [comment]
I had to rebuild `ICU` and `double-conversion`. After that, all good. The TF EasyBlock is complicated enough, but I guess this will bite others as well. Should there be a way to detect shortcomings, and state that "a rebuild of X, Y, and Z in this toolchain is required for this EasyConfig"? Or, optionally, state this without attempts at detecting problem? 

I realise that this is a situation we try very *very* hard to avoid, but it crops up now and then, especially for "modern" applications with a billion dependencies that have extremely specific expectations of their environment.

**Comment by Flamefire** [comment]
I was thinking if EB could do that. E.g. run the sanity check for dependencies again on failure of a build to see if those were updated. 
But anyway the error is usually clear enough in pointing to the dependency. Just telling users to rebuild that an try again should be enough.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/476ee9c9a5d9f7f914171dcdc83ce028 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3308.joltik.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/88d6498cc0921f2f071f655b3a543263 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2407.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/8aefff37191b0f82f3a7ff2e49cc243a for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/0004bc760b26bc4802e229707c2842da for a full test report.

**Comment by boegel** [comment]
Tested on top of https://github.com/easybuilders/easybuild-easyblocks/pull/2166, good to go now imho, thanks @Flamefire!

**Comment by Flamefire** [comment]
Seriously? This gets merged and hours later they release a .1? https://github.com/tensorflow/tensorflow/releases/tag/v2.3.1

@boegel I'd just update this EC if you are ok with that.

**Comment by akesandgren** [comment]
What did you expect? It's Friday, there's always an update of something on a Friday, usually kernels though.

**Comment by terjekv** [comment]
@Flamefire, You have to understand, PyTorch were waiting for you to get this approved, *then* they could release a new version.

**Comment by boegel** [comment]
@Flamefire Updating makes sense, still only in `develop`...

**Comment by Flamefire** [comment]
Updated in https://github.com/easybuilders/easybuild-easyconfigs/pull/11375

**Pull request merged**